### ISSUE TITLE
[BUG] Video API: accessing non-existing frames creates ghost frames

### ISSUE THREAD

**Issue by elsampsa** [voxel51/fiftyone#2238]
### System information

-   **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 20.04.4 LTS
-   **FiftyOne installed from (pip or source)**: pip
-   **FiftyOne version (run `fiftyone --version`)**: 0.16.6
-   **Python version**: 3.8.10

### Describe the problem

I have a video dataset like this:
```
Name:        sfu-hw-objects-v1
Media type:  video
Num samples: 1
Persistent:  True
Tags:        []
Sample fields:
    id:         fiftyone.core.fields.ObjectIdField
    filepath:   fiftyone.core.fields.StringField
    tags:       fiftyone.core.fields.ListField(fiftyone.core.fields.StringField)
    metadata:   fiftyone.core.fields.EmbeddedDocumentField(fiftyone.core.metadata.VideoMetadata)
    media_type: fiftyone.core.fields.StringField
    class_tag:  fiftyone.core.fields.StringField
    name_tag:   fiftyone.core.fields.StringField
Frame fields:
    id:           fiftyone.core.fields.ObjectIdField
    frame_number: fiftyone.core.fields.FrameNumberField
    detections:   fiftyone.core.fields.EmbeddedDocumentField(fiftyone.core.labels.Detections)
```

Frame numbering starts from 10:
```
dataset.distinct("frames.frame_number")
[10,
 11,
 12,
 13,
 14,
 15,
 16,
 ...
509
]
```
To my surprise, I can access frame number 1 (although I shouldn't be able to do this):
```
sample.frames[1]
<FrameView: {'id': None, 'frame_number': 1, 'detections': None}>
```
Now if I do
```
for num in sample.frames:
    print(num)
```
I get:
```
1
10
11
12
13
...
```
So simply accessing ``sample.frames[1]``, instead of rising an ``IndexError`` or the like, *creates* a "ghost" frame object into the Frames list (!)

However, if I try again
```
dataset.distinct("frames.frame_number")
```
The number 1 doesn't show up (presumably because the id of the "ghost" ``Frame`` object is  ``None``).

### Code to reproduce issue

Create any video dataset with a ``Sample`` having ``Frames`` object where the ``frame_number`` members of ``Frame``s start from, say, 10.  Along these lines:
```
f=fo.Frame()
sample.frames.add_frame(frame=f, frame_number=[start indices from 10])
```

To recap:

I don't think if this is a bug or feature.  I think it is a bug.  Creating non-existent Frame objects into Dataset.Sample.Frames when using the [] operator on Dataset.Sample.Frames is very sneaky and can lead to all kinds of unexpected problems.

### Willingness to contribute

The FiftyOne Community encourages bug fix contributions. Would you or another
member of your organization be willing to contribute a fix for this bug to the
FiftyOne codebase?

-   [ ] Yes. I can contribute a fix for this bug independently.
-   [ ] Yes. I would be willing to contribute a fix for this bug with guidance
        from the FiftyOne community.
-   [X ] No. I cannot contribute a bug fix at this time.

**Comment by swheaton** [comment]
intended behavior
